Assets/UI/Panels/citypaneloverview.lua

```diff
@@ -546,50 +546,75 @@ function ViewPanelAmenities( data:table )
-    --Relgion
+
+    -- City-States
+	if GameCapabilities.HasCapability("CAPABILITY_CITY_HUD_AMENITIES_CITY_STATES") then
+        if data.AmenitiesFromCityStates > 0 then
+            CQUI_BuildAmenityBubbleInstance("ICON_GREAT_PERSON_CLASS_WRITER", data.AmenitiesFromCityStates, "LOC_CITY_STATES_TITLE");
+        end
+	end
+    
+    -- Religion
     if GameCapabilities.HasCapability("CAPABILITY_CITY_HUD_AMENITIES_RELIGION") then
         CQUI_BuildAmenityBubbleInstance("ICON_UNITOPERATION_FOUND_RELIGION", data.AmenitiesFromReligion, "LOC_UI_RELIGION_TITLE");
     end
-    --National Parks
+    
+    -- National Parks
     if GameCapabilities.HasCapability("CAPABILITY_CITY_HUD_AMENITIES_NATIONAL_PARKS") then
         CQUI_BuildAmenityBubbleInstance("ICON_UNITOPERATION_DESIGNATE_PARK", data.AmenitiesFromNationalParks, "LOC_PEDIA_CONCEPTS_PAGE_TOURISM_4_CHAPTER_CONTENT_TITLE");
     end
-    --War Weariness
+    
+    -- Starting Era
+	if(data.AmenitiesFromStartingEra > 0) then 
+        CQUI_BuildAmenityBubbleInstance("ICON_GREAT_PERSON_CLASS_SCIENTIST", data.AmenitiesFromStartingEra, "LOC_GAME_START_ERA");
+	end
+    
+    -- Improvements
+    CQUI_BuildAmenityBubbleInstance("ICON_IMPROVEMENT_PASTURE", data.AmenitiesFromImprovements, "LOC_IMPROVEMENT_NAME");
+    
+    -- War Weariness
     if GameCapabilities.HasCapability("CAPABILITY_CITY_HUD_AMENITIES_WAR_WEARINESS") then
-        CQUI_BuildAmenityBubbleInstance("ICON_UNITOPERATION_FORTIFY", (data.AmenitiesLostFromWarWeariness>0 and -data.AmenitiesLostFromWarWeariness or 0), "LOC_PEDIA_CONCEPTS_PAGE_COMBAT_3_CHAPTER_CONTENT_TITLE");
+        if data.AmenitiesLostFromWarWeariness > 0 then 
+            CQUI_BuildAmenityBubbleInstance("ICON_UNITOPERATION_FORTIFY", -data.AmenitiesLostFromWarWeariness, "LOC_PEDIA_CONCEPTS_PAGE_COMBAT_3_CHAPTER_CONTENT_TITLE");
+        end
     end
-    --Bankruptcy
+    
+    -- Bankruptcy
     if GameCapabilities.HasCapability("CAPABILITY_CITY_HUD_AMENITIES_BANKRUPTCY") then
-        CQUI_BuildAmenityBubbleInstance("ICON_NOTIFICATION_TREASURY_BANKRUPT", (data.AmenitiesLostFromBankruptcy>0 and -data.AmenitiesLostFromBankruptcy or 0), "LOC_PEDIA_CONCEPTS_PAGE_GOLD_4_CHAPTER_CONTENT_TITLE");
+        if data.AmenitiesLostFromBankruptcy > 0 then 
+            CQUI_BuildAmenityBubbleInstance("ICON_NOTIFICATION_TREASURY_BANKRUPT", -data.AmenitiesLostFromBankruptcy, "LOC_PEDIA_CONCEPTS_PAGE_GOLD_4_CHAPTER_CONTENT_TITLE");
+        end
     end
 
-    -- CQUI (AZURENCY) TODO : find the best icons for theses bubble
-    -- M4A Fix: data.AmenitiesFromDistricts was not always a number?
-    data.AmenitiesFromDistricts = tonumber(data.AmenitiesFromDistricts) or 0;
-    if (data.AmenitiesFromDistricts > 0) then
-        CQUI_BuildAmenityBubbleInstance("ICON_NOTIFICATION_TREASURY_BANKRUPT", Locale.ToNumber(data.AmenitiesFromDistricts), "LOC_HUD_CITY_AMENITIES_FROM_DISTRICTS");
-    end
+    -- Districts
+	data.AmenitiesFromDistricts = data.AmenitiesFromDistricts or 0;
```

Comment:
if i remember correct the change I made last year, where tonumber() was used was because of some weird bug that was happening here.  I don't remember specifically if including it in this section of code was actually necessary (because changes elsewhere happened to clear up the issue)

Reply:
also, the unmodified file only displays this one if the value is greater than 0 (similar to natural wonders and traits)

Reply:
I don't know what "weird bug" you are referring to. tonumber() is not needed. Line 605 takes care of nil.

Reply:
There are total 15 sources of amenities and only 12 slots. Some of them are conditional. Amenities from Districts are very rare.